compiler/rustc_codegen_cranelift/src/value_and_place.rs

```diff
@@ -437,6 +437,12 @@ impl<'tcx> CPlace<'tcx> {
                 | (types::F32, types::I32)
                 | (types::I64, types::F64)
                 | (types::F64, types::I64) => fx.bcx.ins().bitcast(dst_ty, data),
+
+                // Widen an abstract SSA boolean to something that can be stored in memory
+                (types::B1, types::I8 | types::I16 | types::I32 | types::I64 | types::I128) => {
+                    fx.bcx.ins().bint(dst_ty, data)
+                }
```

Comment:
I would slightly prefer if the `bint` is part of the `raw_eq` impl. That way there is never a `CValue` with a `B1` ssa value, but the `bool` layout for which `clif_type_from_ty` returns `I8`.

Reply:
Sure, can do.  Added a comment above the `unreachable!` too, for the next person implementing things by "what panics now?" 🙃